cmd/pg/wal_restore.go

```diff
@@ -0,0 +1,36 @@
+package pg
+
+import (
+	"github.com/spf13/cobra"
+	"github.com/wal-g/storages/fs"
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/internal"
+)
+
+const (
+	WalRestoreUsage            = "wal-restore target-pgdata source-pgdata"
+	WalRestoreShortDescription = "Restores WAL segments from storage."
+	WalRestoreLongDescription  = "Restores the missing WAL segments that will be needed to perform pg_rewind with storage."
+)
+
+// walRestoreCmd represents the walRestore command
+var walRestoreCmd = &cobra.Command{
+	Use:   WalRestoreUsage,
+	Short: WalRestoreShortDescription,
+	Long:  WalRestoreLongDescription,
+	Args:  cobra.ExactArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
+		targetFolder, err := fs.ConfigureFolder(args[0], nil)
```

Comment:
`storage.Folder` interface is used to interact with the backups/wal archive storage (e.g. S3, Azure, etc). As for the local filesystem operations, you need to use the `os` package.